ext/js/display/display-generator.js

```diff
@@ -356,6 +356,46 @@ export class DisplayGenerator {
         return node;
     }
 
+    /**
+     * @param {import('dictionary').InflectionPossibility} inflectionPossibility
+     * @returns {?HTMLElement}
+     */
+    _createInflectionPossibility(inflectionPossibility) {
+        const {source, inflections} = inflectionPossibility;
+        if (!Array.isArray(inflections) || inflections.length === 0) { return null; }
+        const fragment = this._instantiate('inflection-possibility');
+
+        const sourceIcon = this._getInflectionSourceIcon(source);
+
+        fragment.appendChild(sourceIcon);
+
+        this._appendMultiple(fragment, this._createTermInflection.bind(this), inflections);
+        return fragment;
+    }
+
+    /**
+     * @param {import('dictionary').InflectionSource} source
+     * @returns {HTMLElement}
+     */
+    _getInflectionSourceIcon(source) {
+        const icon = document.createElement('span');
+        icon.classList.add('inflection-source-icon');
```

Comment:
Simplify 3 assignments to just `icon.dataset.inflectionSource = source;` outside of the switch.